File: app/src/main/java/de/markusdamm/pokerapp/data/Gender.java
package de.markusdamm.pokerapp.data;

/**
 * Created by Markus Damm on 30.03.2015.
 */
public class Gender{
    public final static boolean FEMALE = true;
    public final static boolean MALE = false;
    public final static int FEMALE_INT = 1;
    public final static int MALE_INT = 0;
    public final static int BOTH_INT = 2;
    public final static String MALE_STRING = "männlich";
    public final static String FEMALE_STRING = "weiblich";
    public final static String BOTH_STRING = "beide";


    public static int getIntByString(String st){
        switch (st){
            case MALE_STRING:
                return MALE_INT;
            case FEMALE_STRING:
                return FEMALE_INT;
            case BOTH_STRING:
                return BOTH_INT;
            default:
                throw new NullPointerException();
        }

    }

    public static int toInt(boolean b){
        if (b == Gender.MALE){
            return MALE_INT;
        }
        return FEMALE_INT;
    }

    public static boolean toBool(int i){
        if (i == Gender.MALE_INT){
            return MALE;
        }
        return FEMALE;
    }

    public static String getString(boolean b){
        if (b == MALE){
            return MALE_STRING;
        }
        return FEMALE_STRING;
    }

    public static String getStringByInt(int i){
        if (i == MALE_INT){
            return MALE_STRING;
        }
        else{
            if (i == FEMALE_INT){
                return FEMALE_STRING;
            }
            else{
                return BOTH_STRING;
            }
        }
    }

}

File: app/src/main/java/de/markusdamm/pokerapp/data/Gender.kt
package de.markusdamm.pokerapp.data

/**
 * Created by Markus Damm on 30.03.2015.
 */
object Gender {
    const val FEMALE = true
    const val MALE = false
    const val FEMALE_INT = 1
    const val MALE_INT = 0
    const val BOTH_INT = 2
    const val MALE_STRING = "männlich"
    const val FEMALE_STRING = "weiblich"
    const val BOTH_STRING = "beide"

    fun getIntByString(st: String): Int {
        return when (st) {
            MALE_STRING -> MALE_INT
            FEMALE_STRING -> FEMALE_INT
            BOTH_STRING -> BOTH_INT
            else -> throw NullPointerException()
        }
    }

    fun toInt(b: Boolean): Int {
        return if (b == Gender.MALE) {
            MALE_INT
        } else FEMALE_INT
    }

    fun toBool(i: Int): Boolean {
        return if (i == Gender.MALE_INT) {
            MALE
        } else FEMALE
    }

    fun getString(b: Boolean): String {
        return if (b == MALE) {
            MALE_STRING
        } else FEMALE_STRING
    }

    fun getStringByInt(i: Int): String {
        return if (i == MALE_INT) {
            MALE_STRING
        } else {
            if (i == FEMALE_INT) {
                FEMALE_STRING
            } else {
                BOTH_STRING
            }
        }
    }
}